packages/@azure/servicebus/data-plane/test/testUtils.ts

```diff
@@ -193,19 +193,26 @@ export function getReceiverClient(
 }
 
 export async function purge(
-  recieverClient: QueueClient | SubscriptionClient,
-  useSessions?: boolean
+  receiverClient: QueueClient | SubscriptionClient,
+  useSessions?: boolean,
+  sessionId?: string
 ): Promise<void> {
-  const peekedMsgs = await recieverClient.peek();
-  if (!peekedMsgs.length) {
-    return;
-  }
+  let isEmpty = false;
 
-  const receiver = useSessions
-    ? await recieverClient.getSessionReceiver()
-    : recieverClient.getReceiver();
+  while (!isEmpty) {
+    const peekedMsgs = await receiverClient.peek(10);
+    if (peekedMsgs.length === 0) {
+      isEmpty = true;
+    } else {
+      const receiver = useSessions
+        ? await receiverClient.getSessionReceiver({ sessionId: sessionId })
+        : receiverClient.getReceiver();
 
-  receiver.receive(() => Promise.resolve(), (err) => console.log(`Error when purging: ${err}`));
-  await delay(5000);
-  await receiver.close();
+      const msgs = await receiver.receiveBatch(peekedMsgs.length);
+      if (msgs && msgs.length) {
+        await msgs[0].complete();
```

Comment:
Only the first message is getting completed here. You need to loop through `msgs` and complete each of them.

Reply:
Oops, yes - updated code.